services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -158,39 +195,119 @@ class HankeServiceImpl (@Autowired val hankeRepository: HankeRepository) : Hanke
+                    hanke.arvioijat.add(hankeYhteystieto)
+            }
         }
+
+        internal fun createHankeYhteystietoDomainObjectFromEntity(hankeYhteystietoEntity: HankeYhteystietoEntity): HankeYhteystieto {
+            var createdAt: ZonedDateTime? = null
+
+            if (hankeYhteystietoEntity.createdAt != null)
+                createdAt = ZonedDateTime.of(hankeYhteystietoEntity.modifiedAt, TZ_UTC)
+
+            var modifiedAt: ZonedDateTime? = null
+            if (hankeYhteystietoEntity.modifiedAt != null)
+                modifiedAt = ZonedDateTime.of(hankeYhteystietoEntity.modifiedAt, TZ_UTC)
+
+            return HankeYhteystieto(
+                    id = hankeYhteystietoEntity.id,
+                    etunimi = hankeYhteystietoEntity.etunimi,
+                    sukunimi = hankeYhteystietoEntity.sukunimi,
+                    email = hankeYhteystietoEntity.email,
+                    puhelinnumero = hankeYhteystietoEntity.puhelinnumero,
+                    organisaatioId = hankeYhteystietoEntity.organisaatioid,
+                    organisaatioNimi = hankeYhteystietoEntity.organisaationimi,
+                    osasto = hankeYhteystietoEntity.osasto,
+                    createdBy = hankeYhteystietoEntity.createdByUserId?.toString() ?: "",
+                    modifiedBy = hankeYhteystietoEntity.modifiedByUserId?.toString() ?: "",
+                    createdAt = createdAt,
+                    modifiedAt = modifiedAt
+            )
+        }
+    }
+
+    /**
+     * Does NOT copy the id and hankeTunnus fields because one is supposed to find
+     * the HankeEntity instance from the database with those values, and after that,
+     * the values are filled by the database and should not be changed.
+     * Also, version, creatorUserId, createdAt, modifierUserId, modifiedAt, version are not
+     * set here, as they are to be set internally, and depends on which operation
+     * is being done.
+     */
+    internal fun copyNonNullHankeFieldsToEntity(hanke: Hanke, entity: HankeEntity) {
+        hanke.onYKTHanke?.let { entity.onYKTHanke = hanke.onYKTHanke }
+        hanke.nimi?.let { entity.nimi = hanke.nimi }
+        hanke.kuvaus?.let { entity.kuvaus = hanke.kuvaus }
+        // Assuming the incoming date, while being zoned date and time, is in UTC and time value can be simply dropped here.
+        // Note, .toLocalDate() does not do any time zone conversion.
+        hanke.alkuPvm?.let { entity.alkuPvm = hanke.alkuPvm?.toLocalDate() }
+        hanke.loppuPvm?.let { entity.loppuPvm = hanke.loppuPvm?.toLocalDate() }
+        hanke.vaihe?.let { entity.vaihe = hanke.vaihe }
+        hanke.suunnitteluVaihe?.let { entity.suunnitteluVaihe = hanke.suunnitteluVaihe }
+
+        hanke.saveType?.let { entity.saveType = hanke.saveType }
+        hanke.tyomaaKatuosoite?.let { entity.tyomaaKatuosoite = hanke.tyomaaKatuosoite }
+        hanke.tyomaaTyyppi.let { entity.tyomaaTyyppi = hanke.tyomaaTyyppi }
```

Comment:
Maybe: (the "let" has no use for tyyppi at this point, and tyomaakoko can (I think) still be null).
entity.tyomaaTyyppi = hanke.tyomaaTyyppi
hanke.tyomaaKoko?.let { entity.tyomaaKoko = hanke.tyomaaKoko }

Reply:
I just fixed things that gradle build was vaining about. If there is underlying problem in some other way, it is best if you fix it?